ddsrouter_yaml/src/cpp/YamlReader.cpp

```diff
@@ -352,22 +352,22 @@ RealTopic YamlReader::get<RealTopic>(
     std::string type = get_scalar<std::string>(yml, TOPIC_TYPE_NAME_TAG);
 
     // Optional keyed
-    bool keyed;
+    bool keyed = false;
```

Comment:
Assuming the default value of an argument outside the class is a bad practice.
I would leave it as it was before and make an `if else` statement calling one of the 4 possible constructors (I know, it is a shit, this do not happen in python)

Reply:
I've created a new constructor to set the reliable option without setting the keyed option.